src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs

```diff
@@ -1082,6 +1116,26 @@ internal static void AddVtableAdapterTypeForKnownInterface(ITypeSymbol classType
                 }
             }
 
+            if (classType is INamedTypeSymbol namedType && namedType.MetadataName == "ObservableCollection`1")
+            {
+                // ObservableCollection make use of an internal built-in type as part of its
+                // implementation for INotifyPropertyChanged.  Handling that manually here.
+                var genericInterfaces = new List<string>() { "System.Collections.IList", "System.Collections.IEnumerable" };
+                vtableAttributes.Add(
+                    new VtableAttribute(
+                        "System.Collections.Specialized",
+                        false,
+                        "SingleItemReadOnlyList",
+                        ImmutableArray<TypeInfo>.Empty,
+                        "System.Collections.Specialized.SingleItemReadOnlyList",
+                        genericInterfaces.ToImmutableArray(),
+                        ImmutableArray<GenericInterface>.Empty,
+                        false,
+                        false,
+                        false,
+                        "Microsoft.UI.Xaml.Interop.IBindableVector"));

```

Comment:
Leaving a note here, we need to use `typeMapper` to pick the right namespace here for WUX/MUX

Reply:
Fixed.